src/libraries/Microsoft.PowerFx.Core/Public/Config/PowerFxConfig.cs

```diff
@@ -63,6 +84,15 @@ internal void AddEntity(IExternalEntity entity, DName displayName = default)
             _environmentSymbols.Add(entity.EntityName, entity);
         }
 
+        // Sets the "core" builtin functions. This can vary from host to host. 
+        // Overwrite list and set. 
+        internal void SetCoreFunctions(IEnumerable<TexlFunction> functions)
+        {
+            CheckUnlocked();
+
+            _coreFunctions = functions ?? throw new ArgumentNullException(nameof(functions));           
```

Comment:
As a general pattern shouldn't we be checking for null at the start of the function for each parameter and throwing exceptions there, instead of in expressions?

Reply:
Style cop told me to use the throw expression instead of statement. ;)

[https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0016](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0016)

Reply:
Don't we usually pre-check arguments to functions? For example with Contracts.Assert

Reply:
Yeah, that style rule snuck in from the PVA ruleset, and we actually don't have anywhere in our code that's doing this pattern 
if (a == null) throw
var b = a;

since we've mostly used Contracts.Assert/Check for that.

Reply:
I like the coalescing approach, but if you don't feel free to disable that rule in editorconfig